### PR TITLE
feat(http): implement `ok` field in request objects

### PR DESCRIPTION
**Pull request by nlarche**
ISSUES CLOSED: #681

- [X] I added new tests for the issue I fixed/built
- [X] I ran `make test FOO` for the package FOO I'm modifying
- [X] I used `make commit` instead of `git commit`

Hello, 

I only add one test for this feature, tell me if it's enough.

I hope this help.

Nicolas

### CODE REVIEW

File: http/src/interfaces.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export interface RequestOptions {
   category?: string;
   lazy?: boolean;
   responseType?: string;
+  ok?: Function;
```
Comment (jvanbruegge) [inline http/src/interfaces.ts]: I would change that to `(res: SuperAgentResponse) => boolean` to make the typings better

Comment (nlarche) [reply http/src/interfaces.ts]: Ok I change that. I'm not very familiar with typescript yet.

**Comment by staltz** [comment]
Travis tests pass too. In my opinion this is ready to merge. :)

**Comment by jvanbruegge** [comment]
:tada:

**Pull request merged**